Formula/r/rust.rb

```diff
@@ -222,6 +222,22 @@ def post_install
     end
   end
 
+  def caveats
+    <<~EOS
+      If you have installed rustup, the toolchain provided by this package may
+      introduce conflicts.
+
+      To address this issue, please make sure that your rustup installation comes
+      BEFORE this package in $PATH. This is usually done by modifying your shell's
+      configuration (e.g. .bashrc/.zshrc) so that Homebrew's `eval $(brew shellenv)`
+      line is evaluated BEFORE rustup's `. $HOME/.cargo/env` line (the actual syntax
+      might vary).
+
+      You may also want to link this toolchain with `rustup` under the name `system`:
+        rustup toolchain link system "$(brew --prefix rust)"
```

Comment:
```suggestion
      Link this toolchain with `rustup` under the name `system` with:
        rustup toolchain link system "$(brew --prefix rust)"

      If you use rustup, avoid PATH conflicts by following instructions in:
        brew info rustup
```

Then, the actual instructions should be in the `rustup` formula and/or the official Rust documentation (and provide a link to there instead).

As-is the existing text is:
- far too long (the longer the text, the less likely any of it is to be read)
- unclear on what should be added where; there's no specifics

Bear in mind even by considering this (latter paragraph) candidate we're bending our rules; these caveats should be for things that are Homebrew-specific.

Reply:
@MikeMcQuaid Many thanks for your comprehension and this kind suggestion! I will try to add relevant instructions before updating this PR.